symengine/functions.cpp

```diff
@@ -42,6 +42,78 @@ extern umap_basic_basic inverse_cst;
 
 extern umap_basic_basic inverse_tct;
 
+Conjugate::Conjugate(const RCP<const Basic> &arg) : OneArgFunction(arg)
+{
+    SYMENGINE_ASSIGN_TYPEID()
+    SYMENGINE_ASSERT(is_canonical(arg))
+}
+
+bool Conjugate::is_canonical(const RCP<const Basic> &arg) const
+{
+    if (is_a_Number(*arg)) {
+        if (eq(*arg, *ComplexInf)) {
+            return true;
+        }
+        return false;
+    }
+    if (is_a<Constant>(*arg)) {
+        return false;
+    }
+    if (is_a<Mul>(*arg)) {
+        return false;
+    }
+    if (is_a<Pow>(*arg)) {
+        if (is_a<Integer>(*down_cast<const Pow &>(*arg).get_exp())) {
+            return false;
+        }
+    }
+    return true;
+}
+
+RCP<const Basic> Conjugate::create(const RCP<const Basic> &arg) const
+{
+    return conjugate(arg);
+}
+
+RCP<const Basic> conjugate(const RCP<const Basic> &arg)
```

Comment:
For this, can I implement a virtual method inside the functions and call them through this class?